### PR TITLE
Abuse messages with reporter support

### PR DESCRIPTION
**Pull request by Chocobozzz**
Add ability for reporter and moderators of the same instance to send messages on abuses.

My account abuses view:

screen_2020-07-28-11:14:56

Messages modal
screen_2020-07-27-11:35:59

Notification emails:

screen_2020-07-28-09:56:48
screen_2020-07-28-09:57:07

Notifications:
screen_2020-07-28-09:18:19

On a new message, we send a notification to moderators + reporter
On abuse state change, we send a notification to the reporter

### CODE REVIEW

**Comment by kimsible** [comment]
It's a cool feature ! but it's really enormous to review, why not splitting into several steps PRs ?

**Comment by Chocobozzz** [comment]
>  but it's really enormous to review, why not splitting into several steps PRs ?

I'm not expecting reviews, since it's a lot of work to do. It's just an info PR with some screens whose purpose is to detail a little bit how I implemented this roadmap feature.

I you want to review, you can directly check commits. And github allows you to mark files as *Viewed*. Splitting these commits into several PRs will be messy, since many parts are inter-dependant (models, migrations etc).

File: client/src/app/+my-account/my-account-routing.module.ts [hunk 1]
```diff
@@ -162,6 +163,15 @@ const myAccountRoutes: Routes = [
             title: 'Notifications'
           }
         }
+      },
+      {
+        path: 'abuses',
+        component: MyAccountAbusesListComponent,
+        data: {
+          meta: {
+            title: 'My abuses'
```
Comment (rigelk) [inline client/src/app/+my-account/my-account-routing.module.ts:172]: I would suggest replacing "abuse" with "abuse report" as used elsewhere.

File: client/src/app/+my-account/my-account-settings/my-account-notification-preferences/my-account-notification-preferences.component.ts [hunk 2]
```diff
@@ -42,7 +42,9 @@ export class MyAccountNotificationPreferencesComponent implements OnInit {
       newFollow: this.i18n('You or your channel(s) has a new follower'),
       commentMention: this.i18n('Someone mentioned you in video comments'),
       newInstanceFollower: this.i18n('Your instance has a new follower'),
-      autoInstanceFollowing: this.i18n('Your instance auto followed another instance')
+      autoInstanceFollowing: this.i18n('Your instance auto followed another instance'),
```
Comment (rigelk) [inline client/src/app/+my-account/my-account-settings/my-account-notification-preferences/my-account-notification-preferences.component.ts:45]:
```suggestion
      autoInstanceFollowing: this.i18n('Your instance automatically followed another instance'),
```

File: client/src/app/+my-account/my-account-settings/my-account-notification-preferences/my-account-notification-preferences.component.ts [hunk 3]
```diff
@@ -42,7 +42,9 @@ export class MyAccountNotificationPreferencesComponent implements OnInit {
...
+      abuseNewMessage: this.i18n('An abuse received a new message'),
```
Comment (rigelk) [inline client/src/app/+my-account/my-account-settings/my-account-notification-preferences/my-account-notification-preferences.component.ts:46]:
```suggestion
      abuseNewMessage: this.i18n('An abuse report received a new message'),
```

File: client/src/app/+my-account/my-account-settings/my-account-notification-preferences/my-account-notification-preferences.component.ts [hunk 4]
```diff
@@ -42,7 +42,9 @@ export class MyAccountNotificationPreferencesComponent implements OnInit {
...
+      abuseStateChange: this.i18n('One of your abuse has been accepted or rejected by moderators')
```
Comment (rigelk) [inline client/src/app/+my-account/my-account-settings/my-account-notification-preferences/my-account-notification-preferences.component.ts:47]:
```suggestion
      abuseStateChange: this.i18n('One of your abuse reports has been accepted or rejected by moderators')
```

File: client/src/app/+my-account/my-account.component.ts [hunk 5]
```diff
@@ -76,6 +76,11 @@ export class MyAccountComponent implements OnInit {
           routerLink: '/my-account/blocklist/servers',
           iconName: 'peertube-x'
         },
+        {
+          label: this.i18n('My abuses'),
```
Comment (rigelk) [inline client/src/app/+my-account/my-account.component.ts:80]:
```suggestion
          label: this.i18n('My abuse reports'),
```

**Pull request merged**